third_party/move/move-compiler-v2/tests/checking/variants/variants_test_no_parenthesis_ok.exp

```diff
@@ -0,0 +1,35 @@
+// -- Model dump before bytecode pipeline
+module 0x815::m {
+    enum Color {
+        RGB {
+            red: u64,
+            green: u64,
+            blue: u64,
+        }
+        Red,
+        Blue,
+    }
+    enum Generic {
+        Foo {
+            0: #0,
+        }
+        Bar {
+            0: u64,
+        }
+    }
+    private fun test(c: m::Color): bool {
+        test_variants m::Color::Red|RGB(c)
```

Comment:
Does `m::Color::` only apply to the left variant?  Syntactically it makes sense to be able to have a  qualifier separately on each variant, but semantically I suppose they should both be variants of the same enum.  Do we have any test cases where this qualifier is actually needed?  I don't see any.

(This applies to the old parens version, too.)

Reply:
This is the debug print, and it is a bit sloppy. Should be `m::Color::{Red|RGB(c)}`. Yes indeed the compiler only allows variants of the same type, there are test cases in `variants_test_err.move` for this.